libr/bin/pdb/pdb_downloader.c

```diff
@@ -191,6 +151,19 @@ void deinit_pdb_downloader(SPDBDownloader *pd) {
 	pd->download = 0;
 }
 
+static bool is_valid_guid(const char *guid) {
```

Comment:
GUID are 32 chars long, aren't they? I'm not sure this function works correctly.
e.g.:
```
is_valid_guid('12345678901234567890123456789012') should be true
is_valid_guid('123456789012345678901234567890123') should be false
```
I think that currently this function does not return the right values.

Reply:
tested it, function works correctly

Reply:
Mmm, for me it doesn't. I'm using this test, where `is_valid_guid` is copied-pasted from your PR.
```C
#include <stdio.h>
#include <stdbool.h>
#include <ctype.h>

static bool is_valid_guid(const char *guid) {
	if (!guid) {
		return false;
	}
	size_t i;
	for (i = 0; guid[i] && i <= 33 ; i++) {
		if (!isxdigit (guid[i])) {
			return false;
		}
	}
	return i == 33;
}

static void test_s(const char *s) {
	if (is_valid_guid (s)) {
		printf("'%s' is valid guid\n", s);
	} else {
		printf("'%s' is NOT valid guid\n", s);
	}
}

int main(int argc, char **argv) {
	test_s("12345678901234567890123456789012");
	test_s("123456789012345678901234567890123");
	return 0;
}
```
If I run this program, it says:
```
'12345678901234567890123456789012' is NOT valid guid
'123456789012345678901234567890123' is valid guid
```
but the expected output is:
```
'12345678901234567890123456789012' is valid guid
'123456789012345678901234567890123' is NOT valid guid
```
Please let me know if I'm missing something.

Reply:
turns out guid string has appended to it the "age", so assuming it even has a fixed length was wrong, fixed it now

Reply:
mh, that looks weird, but it seems it is something a bit more deeper in the existing code. Guid are 32 chars AFAIK. Nevermind.